compiler/rustc_hir_analysis/src/check/check.rs

```diff
@@ -1032,6 +1032,37 @@ fn check_impl_items_against_trait<'tcx>(
 
     let trait_def = tcx.trait_def(trait_ref.def_id);
 
+    let infcx = tcx
+        .infer_ctxt()
+        // typeck writeback gives us predicates with their regions erased.
+        // As borrowck already has checked lifetimes, we do not need to do it again.
```

Comment:
This comment does not seem correct :thinking: we're looking at the impl signature here which we don't get from typeck. The issue we don't care about regions is that this check relies on impl_wf_check

Reply:
Copy paste ^^ the other case actually makes sense